### DISCUSSION TITLE
Time requirements?

### DISCUSSION THREAD

**Discussion by alexdresko** [microsoft/ML-For-Beginners#75] [Q&A] [answered]
The course is described as being "12 weeks, 24 lessons", but it doesn't give an idea as to how much time is required to complete each lesson. 

I'd like to do the course, but I need to figure out if I've got time in my current schedule.

**Comment by jlooper** [comment, marked as answer]
hi, good question! You should treat the lessons as a virtual lecture room - so a professor would lecture for about an hour (equivalent to reading the lecture and doing two in-class quizzes) and then you'd do homework (knowledge checks/assignments/challenges) - those probably take about one to two hours per lesson. I'd suggest walking through the 4 introductory lessons and seeing how they go. You can use the discussion board as well to track your time!

**Reply by mughal61** [reply]
Hello @jlooper can you please share link for introductory lectures?

**Reply by jlooper** [reply]
Hi, all the lectures are in this repo, and introductory lectures are grouped here: https://github.com/microsoft/ML-For-Beginners/tree/main/1-Introduction

**Reply by AliHaider20** [reply]
Hi @jlooper, where do we attend lectures?

**Reply by 10leoman** [reply]
It seems that this is a self study course(though you can do it with a group as well) all the lectures and content including quizzes assignments etc have been provided in the GitHub! So go ahead and have fun :D

**Reply by jlooper** [reply]
hi @AliHaider20, @10leoman has it right - self-study is the way to go but watch for any announcements on Twitter if we decide to teach parts of this live (I'm @jenlooper on Twitter)
